### DISCUSSION TITLE
Need extra user related data(different from /userinfo response) in token call

### DISCUSSION THREAD

**Discussion by vaibhi1603** [ory/hydra#2412] [Q&A] [answered]
Hi Team, 

I need some extra user related data in token call response, and i can pass that data in accept consent call in id_token in session params. Please help me out here, stuck at this point for long time.
Thanks

**Comment by vinckr** [comment]
Hey, sorry for the late response!
- What exactly are you trying to do, 
- why is it not working, 
- what versions, environment ect. are you using?

**Comment by vinckr** [comment, marked as answer]
From our [FAQ](https://www.ory.sh/hydra/docs/faq)

> How can I store custom data in the client that can be retrieved along with the login request (using the challenge code)?
>I would like send in some metadata to the /oauth2/auth endpoint that can be retrievable by the "challenge code" that Hydra creates.

You can do this using the [request url.](https://github.com/ory/hydra/blob/27dc147a37c1cb98b7beb5428c389545958dd122/consent/types.go#L505)

Both Login and Consent Requests have a request url which contains the original OAuth2 Authorize URL (e.g. /oauth2/auth?client_id=...&your_custom_param=....)
RequestURL is the original OAuth 2.0 Authorization URL requested by the OAuth 2.0 client.
It is the URL which initiates the OAuth 2.0 Authorization Code or OAuth 2.0 Implicit flow.
This URL is typically not needed, but might come in handy if you want to deal with additional request parameters.

**Comment by vinckr** [comment]
Hey @vaibhi1603 ,
Just checking in if your question was answered or if there was still something unclear.
I will mark this as answered for now, but feel free to come back to the discussion any time.

Thanks a ton! 🐝
